buildSrc/src/main/kotlin/exemplar.publishing-conventions.gradle.kts

```
@@ -1,26 +1,22 @@
 plugins {
     id("maven-publish")
+    signing
 }
 
-group = "org.gradle"
+group = "org.gradle.exemplar"
 version = rootProject.version
 
 publishing {
     publications.create<MavenPublication>("mavenJava") {
         artifactId = project.name
         from(components["java"])
     }
+}
 
-    repositories {
-        maven(url = "https://repo.gradle.org/gradle/ext-releases-local") {
-            authentication {
-                credentials {
-                    fun stringProperty(name: String): String? = project.findProperty(name) as? String
-
-                    username = stringProperty("artifactory_user") ?: "nouser"
-                    password = stringProperty("artifactory_password") ?: "nopass"
-                }
-            }
-        }
-    }
+signing {
+    sign(publishing.publications["mavenJava"])
 }
+
+configure<SigningExtension> {
+    useInMemoryPgpKeys(System.getenv("PGP_SIGNING_KEY"), System.getenv("PGP_SIGNING_KEY_PASSPHRASE"))
+}
```

`configure<SigningExtension>` is the same as configuring via the `signing` block.

```suggestion
signing {
    sign(publishing.publications["mavenJava"])
    useInMemoryPgpKeys(System.getenv("PGP_SIGNING_KEY"), System.getenv("PGP_SIGNING_KEY_PASSPHRASE"))
}
```

---

TIL 👍